### ISSUE TITLE
Tensorboard silently failing?

### ISSUE THREAD

**Issue by blazespinnaker** [tensorflow/tensorboard#5479]
Consider Stack Overflow for getting support using TensorBoard—they have
a larger community with better searchability:

https://stackoverflow.com/questions/tagged/tensorboard

Do not use this template for for setup, installation, or configuration
issues. Instead, use the “installation problem” issue template:

https://github.com/tensorflow/tensorboard/issues/new?template=installation_problem.md

To report a problem with TensorBoard itself, please fill out the
remainder of this template.

## Environment information (required)

Please run `diagnose_tensorboard.py` (link below) in the same
environment from which you normally run TensorFlow/TensorBoard, and
paste the output here:

/content/logs/train# tensorboard --inspect --logdir ./======================================================================
Processing event files... (this can take a few minutes)
======================================================================

Found event files in:
./

These tags are in ./:
audio -
histograms -
images -
scalars -
tensor
   keras
======================================================================

Event statistics for ./:
audio -
graph
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
histograms -
images -
scalars -
sessionlog:checkpoint -
sessionlog:start -
sessionlog:stop -
tensor
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
======================================================================

/content/logs/train# ls
events.out.tfevents.1640808802.2b0d5657e854.86.0.v2
events.out.tfevents.1640809278.2b0d5657e854.86.1.v2
events.out.tfevents.1640809346.2b0d5657e854.86.2.v2
events.out.tfevents.1640809720.2b0d5657e854.86.3.v2
/content/logs/train# tensorboard --inspect --logdir ./
======================================================================
Processing event files... (this can take a few minutes)
======================================================================

Found event files in:
./

These tags are in ./:
audio -
histograms -
images -
scalars -
tensor
   keras
======================================================================

Event statistics for ./:
audio -
graph
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
histograms -
images -
scalars -
sessionlog:checkpoint -
sessionlog:start -
sessionlog:stop -
tensor
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
======================================================================

/content/logs/train# tensorboard --inspect --logdir /content/logs
======================================================================
Processing event files... (this can take a few minutes)
======================================================================

Found event files in:
/content/logs/train

These tags are in /content/logs/train:
audio -
histograms -
images -
scalars -
tensor
   keras
======================================================================

Event statistics for /content/logs/train:
audio -
graph
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
histograms -
images -
scalars -
sessionlog:checkpoint -
sessionlog:start -
sessionlog:stop -
tensor
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
======================================================================

/content/logs/train# ls
events.out.tfevents.1640808802.2b0d5657e854.86.0.v2
events.out.tfevents.1640809278.2b0d5657e854.86.1.v2
events.out.tfevents.1640809346.2b0d5657e854.86.2.v2
events.out.tfevents.1640809720.2b0d5657e854.86.3.v2
/content/logs/train# ls  -l
total 49712
-rw-r--r-- 1 root root 12723137 Dec 29 20:13 events.out.tfevents.1640808802.2b0d5657e854.86.0.v2
-rw-r--r-- 1 root root 12723137 Dec 29 20:21 events.out.tfevents.1640809278.2b0d5657e854.86.1.v2
-rw-r--r-- 1 root root 12723137 Dec 29 20:22 events.out.tfevents.1640809346.2b0d5657e854.86.2.v2
-rw-r--r-- 1 root root 12723137 Dec 29 20:28 events.out.tfevents.1640809720.2b0d5657e854.86.3.v2
/content/logs/train# diagnose_tensorboard.py
-bash: diagnose_tensorboard.py: command not found
/content/logs/train# wget https://raw.githubusercontent.com/tensorflow/tensorboard/master/tensorboard/tools/diagnose_tensorboard.py
--2021-12-29 20:49:17--  https://raw.githubusercontent.com/tensorflow/tensorboard/master/tensorboard/tools/diagnose_tensorboard.py
Resolving raw.githubusercontent.com (raw.githubusercontent.com)... 185.199.108.133, 185.199.109.133, 185.199.110.133, ...
Connecting to raw.githubusercontent.com (raw.githubusercontent.com)|185.199.108.133|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 19134 (19K) [text/plain]
Saving to: ‘diagnose_tensorboard.py’

diagnose_tensor 100%[====>]  18.69K  --.-KB/s    in 0.001s  

2021-12-29 20:49:18 (15.6 MB/s) - ‘diagnose_tensorboard.py’ saved [19134/19134]

/content/logs/train# python diagnose_tensorboard.py 
### Diagnostics

<details>
<summary>Diagnostics output</summary>

``````
--- check: autoidentify
INFO: diagnose_tensorboard.py version e43767ef2b648d0d5d57c00f38ccbd38390e38da

--- check: general
INFO: sys.version_info: sys.version_info(major=3, minor=7, micro=12, releaselevel='final', serial=0)
INFO: os.name: posix
INFO: os.uname(): posix.uname_result(sysname='Linux', nodename='2b0d5657e854', release='5.4.144+', version='#1 SMP Tue Dec 7 09:58:10 PST 2021', machine='x86_64')
INFO: sys.getwindowsversion(): N/A

--- check: package_management
INFO: has conda-meta: False
INFO: $VIRTUAL_ENV: None

--- check: installed_packages
INFO: installed: tensorboard==2.7.0
WARNING: no installation among: ['tensorflow', 'tensorflow-gpu', 'tf-nightly', 'tf-nightly-2.0-preview', 'tf-nightly-gpu', 'tf-nightly-gpu-2.0-preview']
INFO: installed: tensorflow-estimator==2.7.0
INFO: installed: tensorboard-data-server==0.6.1

--- check: tensorboard_python_version
INFO: tensorboard.version.VERSION: '2.7.0'

--- check: tensorflow_python_version
INFO: tensorflow.__version__: '2.7.0'
INFO: tensorflow.__git_version__: 'v2.7.0-0-gc256c071bb2'

--- check: tensorboard_data_server_version
INFO: data server binary: '/usr/local/lib/python3.7/dist-packages/tensorboard_data_server/bin/server'
INFO: data server binary version: b'rustboard 0.6.1'

--- check: tensorboard_binary_path
INFO: which tensorboard: b'/usr/local/bin/tensorboard\n'

--- check: addrinfos
socket.has_ipv6 = True
socket.AF_UNSPEC = <AddressFamily.AF_UNSPEC: 0>
socket.SOCK_STREAM = <SocketKind.SOCK_STREAM: 1>
socket.AI_ADDRCONFIG = <AddressInfo.AI_ADDRCONFIG: 32>
socket.AI_PASSIVE = <AddressInfo.AI_PASSIVE: 1>
Loopback flags: <AddressInfo.AI_ADDRCONFIG: 32>
Loopback infos: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('127.0.0.1', 0))]
Wildcard flags: <AddressInfo.AI_PASSIVE: 1>
Wildcard infos: [(<AddressFamily.AF_INET: 2>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('0.0.0.0', 0)), (<AddressFamily.AF_INET6: 10>, <SocketKind.SOCK_STREAM: 1>, 6, '', ('::', 0, 0, 0))]

--- check: readable_fqdn
INFO: socket.getfqdn(): '2b0d5657e854'

--- check: stat_tensorboardinfo
INFO: directory: /tmp/.tensorboard-info
INFO: os.stat(...): os.stat_result(st_mode=16895, st_ino=5373997, st_dev=36, st_nlink=2, st_uid=0, st_gid=0, st_size=4096, st_atime=1640809718, st_mtime=1640809718, st_ctime=1640809718)
INFO: mode: 0o40777

--- check: source_trees_without_genfiles
INFO: tensorboard_roots (1): ['/usr/local/lib/python3.7/dist-packages']; bad_roots (0): []

--- check: full_pip_freeze

fa2==0.3.5
fastai==1.0.61
fastdtw==0.3.4
fastprogress==1.0.0
fastrlock==0.8
fbprophet==0.7.1
feather-format==0.4.1
filelock==3.4.0
firebase-admin==4.4.0
fix-yahoo-finance==0.0.22
Flask==1.1.4
flatbuffers==2.0
folium==0.8.3
future==0.16.0
gast==0.4.0
GDAL==2.2.2
gdown==3.6.4
gensim==3.6.0
geographiclib==1.52
geopy==1.17.0
gin-config==0.5.0
glob2==0.7
google==2.0.3
google-api-core==1.26.3
google-api-python-client==1.12.8
google-auth==1.35.0
google-auth-httplib2==0.0.4
google-auth-oauthlib==0.4.6
google-cloud-bigquery==1.21.0
google-cloud-bigquery-storage==1.1.0
google-cloud-core==1.0.3
google-cloud-datastore==
pytest==3.6.4
python-apt==0.0.0
python-chess==0.23.11
python-dateutil==2.8.2
python-louvain==0.15
python-slugify==5.0.2
python-utils==2.5.6
pytz==2018.9
pyviz-comms==2.1.0
PyWavelets==1.2.0
PyYAML==6.0
pyzmq==22.3.0
qdldl==0.1.5.post0
qtconsole==5.2.1
QtPy==1.11.2
regex==2019.12.20
requests==2.23.0
requests-oauthlib==1.3.0
resampy==0.2.2
retrying==1.3.3
rpy2==3.4.5
rsa==4.8
sacremoses==0.0.46
scikit-image==0.18.3
scikit-learn==1.0.1
scipy==1.4.1
screen-resolution-extra==0.0.0
scs==2.1.4
seaborn==0.11.2
semver==2.13.0
Send2Trash==1.8.0
sentencepiece
transformers==4.15.0
tweepy==3.10.0
typeguard==2.7.1
typing-extensions==3.10.0.2
tzlocal==1.5.1
uritemplate==3.0.1
urllib3==1.24.3
vega-datasets==0.9.0
wasabi==0.8.2
wcwidth==0.2.5
webencodings==0.5.1
Werkzeug==1.0.1
wheel==0.37.0
widgetsnbextension==3.5.2
wordcloud==1.5.0
wrapt==1.13.3
xarray==0.18.2
xgboost==0.90
xkit==0.0.0
xlrd==1.1.0
xlwt==1.3.0
yellowbrick==1.3.post1
zict==2.0.0
zipp==3.6.0

``````

</details>

### Next steps

No action items identified. Please copy ALL of the above output,
including the lines containing only backticks, into your GitHub issue
or comment. Be sure to redact any sensitive information.
/content/logs/train# ^C
/content/logs/train# ^C
/content/logs/train# 
[0] 0:bash*                    "2b0d5657e854" 20:52 29-Dec-21

For browser-related issues, please additionally specify:

  - Browser type and version (e.g., Chrome 64.0.3282.140):
  - Screenshot, if it’s a visual issue:

/content/logs/train# tensorboard --version
2.7.0
/content/logs/train# tensorboard --inspect --logdir /content/logs
======================================================================
Processing event files... (this can take a few minutes)
======================================================================

Found event files in:
/content/logs/train

These tags are in /content/logs/train:
audio -
histograms -
images -
scalars -
tensor
   keras
======================================================================

Event statistics for /content/logs/train:
audio -
graph
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
histograms -
images -
scalars -
sessionlog:checkpoint -
sessionlog:start -
sessionlog:stop -
tensor
   first_step           0
   last_step            0
   max_step             0
   min_step             0
   num_steps            1
   outoforder_steps     []
======================================================================

## Issue description

Running in google collab, fit callback here:

callbacks = [tf.keras.callbacks.TensorBoard(log_dir="/content/logs", 
                              histogram_freq = 1, update_freq=10,embeddings_freq=1),
                     ],

/content/logs/train# ls  -l
total 49712
-rw-r--r-- 1 root root 12723137 Dec 29 20:13 events.out.tfevents.1640808802.2b0d5657e854.86.0.v2
-rw-r--r-- 1 root root 12723137 Dec 29 20:21 events.out.tfevents.1640809278.2b0d5657e854.86.1.v2
-rw-r--r-- 1 root root 12723137 Dec 29 20:22 events.out.tfevents.1640809346.2b0d5657e854.86.2.v2
-rw-r--r-- 1 root root 12723137 Dec 29 20:28 events.out.tfevents.1640809720.2b0d5657e854.86.3.v2

I get a digraph of the model, but nothing tracking scalars like loss / accuracy over time.

The events are written but nothing is getting shown.  Note this is a relatively large model and an epoch takes about 1.5 hours.

**Comment by sanatmpa1** [comment]
@blazespinnaker,

Can you provide a reproducible standalone code to expedite the trouble shooting process? Thanks!

**Comment by bileschi** [comment]
Thanks for including your inspect results.  It seems that the loss / accuracy etc are not getting written to the logs.  It's possible this issue is with TensorBoard, but it could also be with with the `tf.keras.callback`.   Do the loss measurements appear if you use a much smaller test model?

**Comment by blazespinnaker** [comment]
@sanatmpa1 @bileschi 

I reduced the number of rows to a few thousand and it worked, but after it increasing back to 100K, it failed as per above.  

It's not a huge issue, just the silent failure had me chasing it for awhile trying to figure out if I had done something wrong.  I can workaround it by saving values to a file and plotting them separately.

I can't help much further on this issue in terms of testing, though if you have some quick diagnostic values I can check or inspect, I am happy to do that.

Love tensorboard, btw, really great tool.

Cheers

**Comment by bileschi** [comment]
Glad TensorBoard is a helpful part of your development flow!  But sad that you're getting a failure without clear error messaging.  It would be great if we could find out of the failure is happening before or after the transfer from Keras to `tf.summary` ops.  Is there anything noticeable difference in the logs?  Specifically if the threshold of failure is at T, do the logs look different at 90% T vs. 110% of T?
